packages/framework/ai-collab/src/explicit-strategy/agentEditReducer.ts

```
@@ -223,7 +267,14 @@ export function applyAgentEdit(
 				const simpleSchema = fieldSchema as unknown as new (dummy: unknown) => TreeNode;
 				populateDefaults(modification, definitionMap);
 				const constructedModification = new simpleSchema(modification);
-				validator?.(constructedModification);
+				try {
+					validator?.(constructedModification);
+				} catch (error) {
+					if (error instanceof Error) {
+						throw new UsageError(error.message);
+					}
+					throw new Error("The node provided for insertion is not valid");
```

Same as comment above, we shouldn't swallow the original error.